src/System.Xml.XmlSerializer/src/System/Xml/Serialization/XmlAnyElementAttributes.cs

```diff
@@ -16,7 +17,7 @@ namespace System.Xml.Serialization
     /// </devdoc>
     public class XmlAnyElementAttributes : IList
     {
-        private ArrayList _list;
+        private List<object> _list;
```

Comment:
Implementing `IList<XmlAnyElementAttribute>` should be discussed as a separate issue as it'd be a public API addition that would have to go through the API Review process.

Not so sure about using `List<XmlAnyElementAttribute>`... This collection currently allows adding any type of object (e.g. via the non-generic `IList.Add` or indexer), so making it more strict, while technically correct, could break existing code.